src/app/zones/views/ZoneDetails/ZoneDetailsHeader/ZoneDetailsHeader.test.tsx

```diff
@@ -1,137 +1,100 @@
 
 import type { RootState } from "@/app/store/root/types";
 import * as factory from "@/testing/factories";
-import { screen, render, within } from "@/testing/utils";
-
-const mockStore = configureStore();
+import {
+  renderWithBrowserRouter,
+  screen,
+  waitFor,
+  within,
+} from "@/testing/utils";
 
 describe("ZoneDetailsHeader", () => {
-  let initialState: RootState;
-
-  const testZones = factory.zoneState({
-    genericActions: factory.zoneGenericActions({ fetch: "success" }),
-    items: [
-      factory.zone({
-        id: 1,
-        name: "zone-name",
-      }),
-      factory.zone({
-        id: 2,
-        name: "zone2-name",
-      }),
+  let state: RootState;
+  const queryData = {
+    zones: [
+      factory.zone({ id: 1, name: "zone-name" }),
+      factory.zone({ id: 2, name: "zone-name-2" }),
     ],
-  });
+  };
 
   beforeEach(() => {
-    initialState = factory.rootState({
+    state = factory.rootState({
       user: factory.userState({
         auth: factory.authState({
           user: factory.user({ is_superuser: true }),
         }),
       }),
-      zone: testZones,
+      zone: factory.zoneState({
+        genericActions: factory.zoneGenericActions({ fetch: "success" }),
+      }),
     });
   });
 
-  it("displays zone name in header if one exists", () => {
-    const state = initialState;
-    const store = mockStore(state);
-    render(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/zone/1", key: "testKey" }]}
-        >
-          <Routes>
-            <Route element={<ZoneDetailsHeader id={1} />} path="/zone/:id" />
-          </Routes>
-        </MemoryRouter>
-      </Provider>
-    );
+  it("displays zone name in header if one exists", async () => {
+    renderWithBrowserRouter(<ZoneDetailsHeader id={1} />, {
+      state,
+      queryData,
+      route: "/zone/1",
+    });
 
-    const { getByText } = within(screen.getByTestId("section-header-title"));
-    expect(getByText("Availability zone: zone-name")).toBeInTheDocument();
+    await waitFor(() => {
+      expect(screen.getByTestId("section-header-title")).toHaveTextContent(
+        "Availability zone: zone-name"
+      );
+    });
   });
 
-  it("displays not found message if no zone exists", () => {
-    const state = initialState;
-    const store = mockStore(state);
-    render(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/zone/3", key: "testKey" }]}
-        >
-          <Routes>
-            <Route element={<ZoneDetailsHeader id={3} />} path="/zone/:id" />
-          </Routes>
-        </MemoryRouter>
-      </Provider>
+  it("displays not found message if no zone exists", async () => {
+    renderWithBrowserRouter(<ZoneDetailsHeader id={3} />, {
+      state,
+      queryData,
+      route: "/zone/3",
+    });
+
+    const { findByText } = within(
+      await screen.findByTestId("section-header-title")
```

Comment:
Is there a specific reason that you're using `findBy` instead of `getBy` in these tests?

Reply:
findBy is an equivalent of calling `waitFor(() => expect(() => screen.getBy`

https://testing-library.com/docs/dom-testing-library/api-async#findby-queries